packages/google_sign_in/example/lib/main.dart

```diff
@@ -94,18 +97,45 @@ class SignInDemoState extends State<SignInDemo> {
     return null;
   }
 
-  Future<Null> _handleSignIn() async {
-    try {
-      await _googleSignIn.signIn();
-    } catch (error) {
-      print(error);
-    }
+  void _handleSignIn() {
+    _handleGoogleSignIn()
+        .then((GoogleSignInAccount user) => _afterSignIn)
+        .catchError((dynamic error) => _onSignInError);
   }
 
-  Future<Null> _handleSignOut() async {
+  Future<GoogleSignInAccount> _handleGoogleSignIn() async {
+    await _googleSignIn.signIn();
+    return _googleSignIn.currentUser;
+  }
+
+  Future<Null> _handleGoogleSignOut() async {
     _googleSignIn.disconnect();
   }
 
+  void _afterSignIn(GoogleSignInAccount user) {
+    _scaffoldKey.currentState.showSnackBar(new SnackBar(
+        content: new Text("You're signed in " + user.displayName)),
+    );
+  }
+
+  void _onSignInError(dynamic error) {
```

Comment:
This is a moot due to comment above but Dart style guide recommends against dynamic https://www.dartlang.org/guides/language/effective-dart/design#do-annotate-with-object-instead-of-dynamic-to-indicate-any-object-is-accepted